hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -141,20 +141,20 @@ public class HoodieTableMetaClient implements Serializable {
    * Instantiate HoodieTableMetaClient.
    * Can only be called if table already exists
    */
-  protected HoodieTableMetaClient(StorageConfiguration<?> conf, String basePath, boolean loadActiveTimelineOnLoad,
+  protected HoodieTableMetaClient(HoodieStorage storage, String basePath, boolean loadActiveTimelineOnLoad,
```

Comment:
HoodieStorage is not serializable. and is marked as transient in HoodieTableMetaClient. 
accounting for this, are we good wrt changes in this patch? 
if we pass HoodieTableMetaClient from driver to exe, how does exec reconstruct the HoodieStorage?

Reply:
or in L153, should we try to call getStorage() just after 
```
this.storage = storage;
```
bcoz, on driver this could be null right.

Reply:
may be we are leaving it lazy. when constructing for the first time in executor, storage could be null. but later if getStorage() is invoked, it might instantiate HoodieStorage. 

just wanted to ensure my understanding is right and there are no gaps here.

Reply:
Within this PR, we don't change the place where the meta client is instantiated within the executor. `HoodieTableMetaClient` still holds the `storageConf` instance which is serializable; so the meta client still has the `storageConf` after deserialization on the executor (sent from driver), and the `HoodieStorage` is instantiated first by leveraging the storage configuration, i.e., `getStorage()` instantiates `this.storage` if it is null.

Reply:
We can add extra protection to check nulls.

Reply:
got it. thats what I wanted to confirm.